File: packages/typescript/src/tools/ScrollTool.ts
import { BaseDriver } from "../drivers/BaseDriver.ts";
import { BaseTool } from "./BaseTool.ts";
import { field, type FieldMetadata } from "./Field.ts";

export class ScrollTool extends BaseTool {
  static description = "Scroll to an element.";
  static fields: FieldMetadata[] = [
    field({
      name: "id",
      type: "integer",
      description: "Element identifier (ID)",
    }),
  ];

  id: number;

  constructor(args: { id: number }) {
    super();
    this.id = args.id;
  }

  async invoke(driver: BaseDriver): Promise<void> {
    await driver.scrollTo(this.id);
  }
}

File: packages/python/src/alumnium/tools/scroll_tool.py
from pydantic import Field

from ..drivers.base_driver import BaseDriver
from .base_tool import BaseTool


class ScrollTool(BaseTool):
    """Scroll to an element."""

    id: int = Field(description="Element identifier (ID)")

    def invoke(self, driver: BaseDriver):
        driver.scroll_to(self.id)
